File: Greedy/src/main/java/Solution122.java
public class Solution122 {

    public int maxProfit(int[] prices) {
        int n = prices.length;

        int ans = 0;
        for (int i = 1; i < n; i++) {
            ans = Math.max(prices[i] - prices[i - 1], 0) + ans;
        }

        return ans;
    }

}

File: Greedy/src/main/python/solution122.py
# https://leetcode-cn.com/problems/best-time-to-buy-and-sell-stock-ii/
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n: int = len(prices)
        ans: int = 0
        for i in range(1, n):
            ans = max(prices[i] - prices[i - 1], 0) + ans

        return ans
